warehouse/query-core/src/main/java/datawave/query/jexl/visitors/validate/ValidateBoundedRangeVisitor.java

```diff
@@ -0,0 +1,52 @@
+package datawave.query.jexl.visitors.validate;
+
+import static datawave.query.jexl.nodes.QueryPropertyMarker.MarkerType.BOUNDED_RANGE;
+
+import org.apache.commons.jexl3.parser.ASTAndNode;
+import org.apache.commons.jexl3.parser.JexlNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import datawave.query.jexl.JexlASTHelper;
+import datawave.query.jexl.LiteralRange;
+import datawave.query.jexl.nodes.QueryPropertyMarker;
+import datawave.query.jexl.nodes.QueryPropertyMarker.Instance;
+import datawave.query.jexl.visitors.ShortCircuitBaseVisitor;
+
+/**
+ * Verifies the lower bound sorts before the upper bound for all bounded ranges.
+ * <p>
+ * This visitor must be run after bounded range expansion due to how normalization happens.
+ */
+public class ValidateBoundedRangeVisitor extends ShortCircuitBaseVisitor {
```

Comment:
Will you be adding tests for this visitor?

Reply:
added in next commit